src/lib/data/github.ts

```diff
@@ -32,25 +34,87 @@ export class GitHubCollector {
     }
 
     try {
-      const response = await fetch(`https://api.github.com/repos/${repoInfo.owner}/${repoInfo.repo}`);
-      if (!response.ok) {
-        throw new Error(`GitHub API returned ${response.status}`);
+      // Fetch repository data
+      const repoResponse = await fetch(`https://api.github.com/repos/${repoInfo.owner}/${repoInfo.repo}`);
```

Comment:
you're fetching this in basically every metric, make sure when you split out the fetch stuff you don't call this multiple times